src/Views/AppInfoView.vala

```diff
@@ -281,32 +281,48 @@ namespace AppCenter.Views {
             new Thread<void*> ("content-loading", () => {
                 app_version.label = package.get_version ();
 
-                string url = null;
-                uint max_size = 0U;
-
                 if (screenshots.length == 0) {
                     return null;
                 }
 
+                Gee.Deque<Gtk.Widget> screenshot_widgets = new Gee.LinkedList<Gtk.Widget> ();
```

Comment:
Why did you declare `screenshot_widgets` as a Gee.Deque if you are creating a new Gee.LinkedList?